### PR TITLE
Added Java Implementation of Disjoint Sets

### LINKED ISSUE #1287: Disjoint sets in Java

### PR DESCRIPTION
**Pull request by dibyaaaaax**
Fixes #1287 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Java implementation of Disjoint sets
- 

#### Languages Used:
- Java
- 

#### Files Added:
-  Disjoint_Sets/Disjoint_Sets.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Disjoint_Sets/Disjoint_Sets.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+public class Disjoint_Sets {
+    //declare global arrays to store the parent of each element and size of each set
+    static int[] parent;
+    static int[] size;
+    
+    //function that initially defines the parent of all element as itself.
+    static void makeSet(int n) {
+        for (int i=0;i<n+1;i++) {
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: Add a space after ``;`` and always follow coding conventions (https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html).
Add space around operators according to coding conventions throughout this program.

File: Disjoint_Sets/Disjoint_Sets.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            parent[i]=i;
+            size[i]=1;
+        }
+        return;
+    }
+    
+
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: Remove extra lines like this one (following coding conventions).

File: Disjoint_Sets/Disjoint_Sets.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        System.out.println(find(1)+ "  " + find(4));
+        
+        //check if 2 and 6 belong to the same set (expected output: 2  2)
+        System.out.println(find(2)+ "  " + find(6));
+        
+        //check if 1 and 2 belong to the same set (expected output: 1  2)
+        System.out.println(find(1)+ "  " + find(2));
+    }
+
+}
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: Add a sample input-output at the end of file.

File: Disjoint_Sets/Disjoint_Sets.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        union(2,5);
+        union(5,6); 
+        
+         //check if 1 and 4 belong to the same set (expected output: 1  1)
+        System.out.println(find(1)+ "  " + find(4));
+        
+        //check if 2 and 6 belong to the same set (expected output: 2  2)
+        System.out.println(find(2)+ "  " + find(6));
+        
+        //check if 1 and 2 belong to the same set (expected output: 1  2)
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: In all comments, add a space after **//**.

**Comment by marwahmanbir** [comment]
@dibyaaaaax I see that the issue #1287 is not assigned to you yet, you can only work on issues that are assigned to you.

File: Disjoint_Sets/Disjoint_Sets.java (hunk 4 above)
Comment (dibyaaaaax) [reply Disjoint_Sets/Disjoint_Sets.java]: @marwahmanbir 
Done. Please review.

Also, sorry for not waiting for the issue to be assigned to me.

File: Disjoint_Sets/Disjoint_Sets.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+
+
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: Remove this extra line.

File: Disjoint_Sets/Disjoint_Sets.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+public class Disjoint_Sets {
+	static int[] parent;
+	static int[] size;
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: @dibyaaaaax I guess the indentation is messed up in the new commit. Please fix this.
You have to add 4 spaces in place of single indentation, 8 spaces in place of double, and so on...
You can do this directly from the IDE / text editor's quick settings.

File: Disjoint_Sets/Disjoint_Sets.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    
+    // function to merge two elements such that both of them belong to the same set
+    static void union(int a, int b) {
+        int parent_a = find(a);
+        int parent_b = find(b);
+        
+        if (parent_a != parent_b) {
+            if (size[parent_a] < size[parent_b]) {
+                // swap a and b
```
Comment (marwahmanbir) [inline Disjoint_Sets/Disjoint_Sets.java]: Add a blank line before this comment.

File: Disjoint_Sets/Disjoint_Sets.java (hunk 6 above)
Comment (dibyaaaaax) [reply Disjoint_Sets/Disjoint_Sets.java]: Done.

**Pull request merged**